File: Bridge/Pasternak/Shape/Triangle.java
package Inheritance.Shape;


import Inheritance.Color.Color;

public class Triangle extends Shape {
    private double firstSide;
    private double secondSide;
    private double thirdSide;

    public Triangle(double firstSide, double secondSide, double thirdSide, Color color) {
        super(color);
        this.firstSide = firstSide;
        this.secondSide = secondSide;
        this.thirdSide = thirdSide;
    }

    public double getPerimeter() {
        return this.firstSide + this.secondSide + this.thirdSide;
    }

    public double getArea() {
        double p = getPerimeter()/2;
        return Math.sqrt(p  * (p - this.firstSide) * (p - this.secondSide) * (p - this.thirdSide));
    }

    public String toString() {
        return "Triangle{firstSide=" + this.firstSide + ", secondSide=" + this.secondSide + ", thirdSide=" + this.thirdSide + ", perimeter=" + this.getPerimeter() + ", area=" + this.getArea() + ", color=" + this.color + "}";
    }
}

File: Bridge/Trushchak-03/Bridge/Bridge/Triangle.cs
using System;

namespace Bridge
{
    class Triangle : Shape
    {
        private double side1;
        private double side2;
        private double side3;

        public Triangle(double side1, double side2, double side3, IColor color) : base(color)
        {
            this.side1 = side1;
            this.side2 = side2;
            this.side3 = side3;
        }

        public override double GetArea()
        {
            try
            {
                
                if (side1 <= 0 || side2 <= 0 || side3 <= 0)
                    throw new ArgumentException("All sides of a triangle must be positive.");

                
                double semiPerimeter = (side1 + side2 + side3) / 2;
                return Math.Sqrt(semiPerimeter * (semiPerimeter - side1) * (semiPerimeter - side2) * (semiPerimeter - side3));
            }
            catch (ArgumentException ex)
            {
                Console.WriteLine($"Error calculating area: {ex.Message}");
                return 0;
            }
        }

        public override double GetPerimeter()
        {
            try
            {
                
                if (side1 <= 0 || side2 <= 0 || side3 <= 0)
                    throw new ArgumentException("All sides of a triangle must be positive.");

                return side1 + side2 + side3;
            }
            catch (ArgumentException ex)
            {
                Console.WriteLine($"Error calculating perimeter: {ex.Message}");
                return 0;
            }
        }

        public override void Draw()
        {
            Console.WriteLine($"Drawing a {color.GetColor()} triangle with sides {side1}, {side2}, {side3}");
        }
    }
}
